frame/pallet-xvm/src/evm.rs

```diff
@@ -26,8 +26,10 @@ where
             "Start EVM XVM: {:?}, {:?}, {:?}",
             from, to, input,
         );
-        let value = U256::from(0u64);
-        let max_fee_per_gas = U256::from(3450898690u64);
+        let value = U256::zero();
+
+        // Tells the EVM executor that no fees should be charged for this execution.
+        let max_fee_per_gas = U256::zero();
```

Comment:
Is it really works? I guess EVM has minimal fee and return an error if providen fee is less.

Reply:
I believe it will work.

Calculated initial fee will be `zero`.

```
	fn withdraw_fee(who: &H160, fee: U256) -> Result<Self::LiquidityInfo, Error<T>> {
		if fee.is_zero() {
			return Ok(None);   <===========
		}
               ...
```

Followed by:
```
fn correct_and_deposit_fee(
		who: &H160,
		corrected_fee: U256,
		base_fee: U256,
		already_withdrawn: Self::LiquidityInfo,
	) -> Self::LiquidityInfo {
		if let Some(paid) = already_withdrawn {
                    .....
		}
		None   <========
	}
```

So the whole payment will be a no-op.

Reply:
@akru it seems that is correct - this step happens in transaction validation.

We'll have to rethink how we approach this - probably calling `EVM` pallet isn't the way to go.